dali/operators/imgcodec/decoder.cc

```diff
@@ -0,0 +1,411 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/imgcodec/decoder.h"
+#include "dali/pipeline/operator/common.h"
+#include "dali/pipeline/operator/operator.h"
+#include "dali/pipeline/data/views.h"
+#include "dali/imgcodec/image_decoder.h"
+
+#include "dali/operators/imgcodec/operator_utils.h"
+
+namespace dali {
+namespace imgcodec {
+
+DALI_SCHEMA(ImgcodecDecoderAttr)
+.AddOptionalArg("output_type",
+      R"code(The color space of the output image.
+
+Note: When decoding to YCbCr, the image will be decoded to RGB and then converted to YCbCr,
+following the YCbCr definition from ITU-R BT.601.
+)code",
+      DALI_RGB)
+  .AddOptionalArg("hybrid_huffman_threshold",
```

Comment:
```suggestion
  .AddOptionalArg("adjust_oridentation",
```
...and default to `true`.

Reply:
Done